nexusutils/dataconverter/readers/em_spctrscpy/utils/em_nexus_base_classes.py

```diff
@@ -39,44 +39,44 @@ def __init__(self,
                  value=None,
                  **kwargs):
         if name is not None:
-            assert name != '', 'Argument name needs to be a non-empty string !'
+            assert name != "", "Argument name needs to be a non-empty string !"
         if unit is not None:
-            assert unit != '', 'Argument unit needs to be a non-empty string !'
-        assert dtype is not None, 'Argument dtype must not be None !'
+            assert unit != "", "Argument unit needs to be a non-empty string !"
+        assert dtype is not None, "Argument dtype must not be None !"
         if dtype is not None:
             assert isinstance(dtype, type), \
-                'Argument dtype needs a valid, ideally numpy, datatype !'
+                "Argument dtype needs a valid, ideally numpy, datatype !"
         # ##MK::if value is not None:
-        self.is_a = 'NXobject'
+        self.is_a = "NXobject"
         self.is_attr = False  # if True indicates object is attribute
-        self.doc = ''  # docstring
+        self.doc = ""  # docstring
         self.name = name  # name of the field
         self.unit = unit  # not unit category but actual unit
-        # use special values 'unitless' for NX_UNITLESS (e.g. 1) and
-        # 'dimensionless' for NX_DIMENSIONLESS (e.g. 1m / 1m)
+        # use special values "unitless" for NX_UNITLESS (e.g. 1) and
+        # "dimensionless" for NX_DIMENSIONLESS (e.g. 1m / 1m)
         self.dtype = dtype  # use np.dtype if possible
         if value is None or dtype is str:
-            self.unit = 'unitless'
+            self.unit = "unitless"
         if value is not None:
             self.value = value
         else:
             self.value = None
         # value should be a numpy scalar, tensor, or string if possible
-        if 'is_attr' in kwargs.keys():
-            assert isinstance(kwargs['is_attr'], bool), \
-                'Kwarg is_attr needs to be a boolean !'
-            self.is_attr = kwargs['is_attr']
+        if "is_attr" in kwargs:
+            assert isinstance(kwargs["is_attr"], bool), \
+                "Kwarg is_attr needs to be a boolean !"
+            self.is_attr = kwargs["is_attr"]
 
     def print(self):
```

Comment:
Consider using Python's built in __repr__ or __str__. They allow you to control the output of `print(YOUR_OBJECT) `

Reply:
Done